### PR TITLE
Improve game data flow and charts

### PR DESCRIPTION
**Pull request by jnwltr**
- stream complete data (state array) from game service
- byproduct: fix disconnected subscribers after game restart
- buffer the last state array for late subscribers
- use data incrementally (new items in the array) in charts
- display immunity chart as stacked with legend

### CODE REVIEW

**Comment by jnwltr** [comment]
Screenshot 2021-01-09 at 18 51 32

**Review by daeglin — APPROVED** [review]

File: frontend/src/app/game/status-display/status-display.component.ts [hunk 1]
```diff
@@ -11,14 +11,18 @@ import {GameService} from '../game.service';
   styleUrls: ['./status-display.component.scss'],
 })
 export class StatusDisplayComponent {
+  lastState$ = this.gameService.gameState$.pipe(
+    filter(states => states.length > 0),
+    map(states => states[states.length - 1]),
```
Comment (JaroslavSoldat) [inline frontend/src/app/game/status-display/status-display.component.ts:16]: For reset view when game restarted could be better
`map(states => states?.length ? states[states.length - 1] : undefined),`

Next pipes must be edited to correctly process `undefined`.

File: frontend/src/app/game/graphs/graphs.component.ts [hunk 2]
```diff
@@ -31,160 +31,183 @@ export class GraphsComponent implements AfterViewInit {
...
     for (const mitigation of Object.keys(MitigationsService.mitigationsI18n)) {
       this.mitigationsService.formGroup.get(mitigation)?.valueChanges.pipe(
         untilDestroyed(this),
       ).subscribe((value: any) => {
-        // TODO add multiple mitigations at a same time
-        this.currentMitigation = this.mitigationsService.getLabel(
+        const label = this.mitigationsService.getLabel(
           mitigation as keyof typeof MitigationsService.mitigationsI18n, value);
+        if (this.currentMitigation) this.currentMitigation += '\n' + label;
+        else this.currentMitigation = label;
```
Comment (JaroslavSoldat) [inline frontend/src/app/game/graphs/graphs.component.ts:210]: Consider to use
`this.currentMitigation = [this.currentMitigation, label].filter(Boolean).join('\n');`

File: frontend/src/app/game/game/game.component.ts [hunk 3]
```diff
@@ -30,10 +30,11 @@ export class GameComponent {
 
     // send current game result into outro
     gameService.gameState$.pipe(
-      map(state => state.stats),
-      map(stats => ({
-        dead: stats.deaths.total,
-        cost: stats.costs.total,
+      filter(states => states.length > 0),
+      map(states => states[states.length - 1]),
```
Comment (JaroslavSoldat) [inline frontend/src/app/game/game/game.component.ts:34]: For reset result when game restarted could be better
`map(states => states?.length ? states[states.length - 1] : undefined),`

Next `map` must be edited to correctly process undefined.

**Review by JaroslavSoldat — COMMENTED** [review]
Great job! 👍🏻 

Dunno how game restart is processed. I assume that the next value of `gameService.gameState$` after the reset is `[]`.

**Pull request merged**